apps/nestjs-backend/src/features/view/open-api/view-open-api.service.ts

```diff
@@ -250,28 +250,89 @@ export class ViewOpenApiService {
     });
   }
 
-  async updateViewOrder(tableId: string, viewId: string, orderRo: IViewOrderRo) {
-    const { order } = orderRo;
-
+  /**
+   * shuffle view order
+   */
+  async shuffle(tableId: string) {
     const views = await this.prismaService.view.findMany({
-      select: { order: true, id: true },
       where: { tableId, deletedTime: null },
+      select: { id: true, order: true },
+      orderBy: { order: 'asc' },
     });
 
-    const curView = views.find(({ id }) => id === viewId);
+    this.logger.log(`lucky view shuffle! ${tableId}`, 'shuffle');
 
-    if (!curView) {
-      throw new BadRequestException('View not found in the table');
-    }
+    await this.prismaService.$tx(async () => {
```

Comment:
```
await this.prismaService.$tx(async () => {
  await Promise.all(views.map(async (view, i) => {
    await this.viewService.updateViewByOps(tableId, view.id, [
      ViewOpBuilder.editor.setViewProperty.build({
        key: 'order',
        newValue: i,
        oldValue: view.order,
      }),
    ]);
  }));
});
```
I think it should be like this. It would be better.

Reply:
Thanks for review

A large number of concurrent operations in a single transaction will not yield much incremental benefit, and this is not a bottleneck

Reply:
I think it is a bottleneck, It will await sequence updateViewByOps
and when using Promise.all,  it will await the longest updateViewByOps

Reply:
First off, from the user's perspective, there won't be a large number of views, so the speed difference is almost imperceptible. However, if there really is a large quantity, the inherent overhead of numerous promises could actually slow things down. In another feature we implemented, we adopted the method of copying to a temporary table for bulk updates, which, after testing, resulted in a more than tenfold improvement in speed and is currently the fastest method we've found.

Another point worth mentioning is that we also support sqlite, which does not support concurrent writing. Attempting this in end-to-end (e2e) tests with sqlite has led to some strange errors.